frontend/src/views/AccountView.vue

```diff
@@ -16,22 +16,23 @@
           { field: 'Full Name', value: [user.full_name] },
           { field: 'Email', value: [user.email] },
         ]" />
-      <SectionBox
+      <CredentialsBox
         :header="'Credentials'"
-        :content= credentialsContent
-        @click="toggleSecret"
+        :clientId="user.clientId"
+        :clientSecret="secretValue"
         data-test="credentials"
         :key="showSecretValue"
-      />
-      <button @click="generateSecret" type="submit">
-        Generate Secret
-      </button>
+        @display-secret= this.onToggleSecret
+        @generate-secret = this.onGenerateSecret
+        :onSecretGenerated="updateSecretValue"
```

Comment:
Was things not re-rendering and that is why it is being provided as a call-back function?  https://github.com/uab-cgds-worthey/rosalution/pull/58/files#diff-b92c9527c7cb48e5062f9737c31fa327eaeba23323b203d235c40417868ea050R75 when calling this function on line 75 within the component, it is making an assumption that the secret was generated correctly.  So the component could incorrectly make the assumption that the that secret was successfully generated.    You could make a prop that is 'secretExists' that could get changed to true when the secret updates, but it should already be re-rendering from the secret.